src/main.cc

```diff
@@ -41,36 +42,46 @@ struct Stats {
   Scalar scattered_flux; // Watts / m^2
 };
 
+int parse_time(std::string& time_string, struct tm& tv, int& tz){
```

Comment:
Could you change this to the following? It's a little bit easier for me to understand than these mutable references.
```suggestion
std::optional<std::pair<struct tm, int>> parse_time(const std::string &time_string) {
```